From cd2e04ce5a2f6c81207f5c497cbab85c16429daf Mon Sep 17 00:00:00 2001
From: Nithin Dabilpuram <ndabilpuram@marvell.com>
Date: Fri, 28 Jun 2019 15:59:07 +0530
Subject: [PATCH 263/767] octeontx2-af: Fix inner l4 layer id

commit 68c5eadb758fd20ae71f7e1471c3f27db8b658e0 from
git@git.assembla.com:cavium/WindRiver.linux.git

Current NPC profile has designated Inner L4 layer as LH. This fixes
RSS flow key alg preperation to use LH instead of LG.

Change-Id: If14dad7ffa1f1300bd6f412394e59fd451610f74
Signed-off-by: Nithin Dabilpuram <ndabilpuram@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/11758
Reviewed-by: Naveen Mamindlapalli <Naveen.Mamindlapalli@cavium.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
index 975670a8ce25..7d9820062cdf 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
@@ -2625,7 +2625,7 @@ static int set_flowkey_fields(struct nix_rx_flowkey_alg *alg, u32 flow_cfg)
 			if (key_type == NIX_FLOW_KEY_TYPE_INNR_TCP ||
 			    key_type == NIX_FLOW_KEY_TYPE_INNR_UDP ||
 			    key_type == NIX_FLOW_KEY_TYPE_INNR_SCTP)
-				field->lid = NPC_LID_LG;
+				field->lid = NPC_LID_LH;
 			field->bytesm1 = 3; /* Sport + Dport, 4 bytes */
 
 			/* Enum values for NPC_LID_LD and NPC_LID_LG are same,
-- 
2.31.1

